app/src/features/mocks/screens/mocksList/components/MocksList/components/MocksListContentHeader/MocksListContentHeader.tsx

```diff
@@ -1,31 +1,41 @@
-import React from "react";
+import React, { useMemo } from "react";
 import { useSelector } from "react-redux";
 import { CloudUploadOutlined, PlusOutlined } from "@ant-design/icons";
-import { ButtonProps } from "antd";
+import { Badge, ButtonProps } from "antd";
 import { AuthConfirmationPopover } from "components/hoc/auth/AuthConfirmationPopover";
-import { ContentListHeader } from "componentsV2/ContentList";
-import { MockType } from "components/features/mocksV2/types";
+import { ContentListHeader, ContentListHeaderProps } from "componentsV2/ContentList";
+import { MockTableHeaderFilter, MockType, RQMockMetadataSchema } from "components/features/mocksV2/types";
 import { RQButton } from "lib/design-system/components";
 import { SOURCE } from "modules/analytics/events/common/constants";
 import { getUserAuthDetails } from "store/selectors";
 import { MdOutlineCreateNewFolder } from "@react-icons/all-files/md/MdOutlineCreateNewFolder";
+import { MdOutlineStarOutline } from "@react-icons/all-files/md/MdOutlineStarOutline";
+import { isRecordMock } from "../MocksTable/utils";
 
 interface Props {
   mockType?: string;
+  allrecords: RQMockMetadataSchema[];
+  mocks: RQMockMetadataSchema[];
   handleUploadAction?: () => void;
   handleCreateNew: () => void;
   handleCreateNewCollection?: () => void;
   searchValue?: string;
   setSearchValue?: (s: string) => void;
+  filter: MockTableHeaderFilter;
+  setFilter: (filter: MockTableHeaderFilter) => void;
 }
 
 export const MocksListContentHeader: React.FC<Props> = ({
+  mocks,
```

Comment:
We don't need to pass mocks separately here. Just pass records only and do the computation here.

Also rename `allrecords` to `records`. Records implicitly means allRecords